### PR TITLE
A few minor `net` component cleanups.

### PR DESCRIPTION
**Pull request by frewsxcv**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/10725)

### CODE REVIEW

File: components/net/cookie.rs [hunk 1]
```diff
@@ -66,7 +66,7 @@ impl Cookie {
 
         // Step 7
         let mut path = cookie.path.unwrap_or("".to_owned());
-        if path.is_empty() || path.as_bytes()[0] != b'/' {
+        if path.chars().nth(0) != Some('/') {
```
Comment (KiChjang) [inline components/net/cookie.rs]: `.next()` instead of `.nth(0)`?

File: components/net/cookie.rs [hunk 2]
```diff
@@ -96,7 +96,7 @@ impl Cookie {
     // http://tools.ietf.org/html/rfc6265#section-5.1.4
     pub fn default_path(request_path: &str) -> &str {
         // Step 2
-        if request_path.is_empty() || !request_path.starts_with("/") {
+        if request_path.chars().nth(0) != Some('/') {
```
Comment (KiChjang) [inline components/net/cookie.rs]: Ditto.

File: components/net/cookie.rs (hunk 1 above)
Comment (frewsxcv) [inline components/net/cookie.rs]: Fixed in latest force push

**Comment by frewsxcv** [comment]
I accidentally applied the `next()` change to the last commit instead of the first one.

@bors-servo r=KiChjang

**Comment by KiChjang** [comment]
@bors-servo retry
* #10698, #10730

**Pull request merged**